### PR TITLE
Improve docker image

### PR DESCRIPTION
**Pull request by babygoat**

### CODE REVIEW

**Comment by babygoat** [comment]
@nickhsine @YuCJ @taylrj  
I made some changes to the CI  and docker image construction flows, which should reduce the size during the deployment and speed up the building flow.

For some quick validations, the deployment time can be saved more than 50% and size could be reduced to 25% with respect to the original status.

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+*.md
+*.txt
+*.png
+Makefile
+package.json
+build-babel-preset.js
+build-pm2-processes.js
+yarn.lock
+
+# Binaries
```
Comment (nickhsine) [inline .dockerignore]: remove `# Binaries` ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -15,17 +15,31 @@ jobs:
...
 
       - save_cache:
           paths:
             - node_modules
-          key: v1-dependencies-{{ checksum "yarn.lock" }}
+          key: v1-dependencies-build-{{ checksum "yarn.lock" }}
 
       - run: make build RELEASE_BRANCH=${CIRCLE_BRANCH}
 
+      - run: rm -rf node_modules
```
Comment (nickhsine) [inline .circleci/config.yml]: Is it ok to remove `node_modules` folder?

Comment (babygoat) [reply .circleci/config.yml]: The packages for runtime are installed in the following steps.

**Comment by nickhsine** [comment]
LGTM

File: .dockerignore (hunk 1 above)
Comment (babygoat) [reply .dockerignore]: This should be executables. But I was wondering why these two files are executable.

**Comment by babygoat** [comment]
updated

**Pull request merged**